src/backend/distributed/commands/trigger.c

```diff
@@ -203,45 +214,602 @@ get_relation_trigger_oid_compat(HeapTuple heapTuple)
+ */
+List *
+PostprocessAlterTriggerRenameStmt(Node *node, const char *queryString)
+{
+	RenameStmt *renameTriggerStmt = castNode(RenameStmt, node);
+	Assert(renameTriggerStmt->renameType == OBJECT_TRIGGER);
+
+	RangeVar *relation = renameTriggerStmt->relation;
+
+	bool missingOk = false;
+	Oid relationId = RangeVarGetRelid(relation, ALTER_TRIGGER_LOCK_MODE, missingOk);
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	ErrorIfUnsupportedAlterTriggerRenameCommand(renameTriggerStmt);
+
+	if (IsCitusLocalTable(relationId))
+	{
+		/* use newname as standard process utility already renamed it */
+		char *triggerName = renameTriggerStmt->newname;
+		return CitusLocalTableTriggerCommandDDLJob(relationId, triggerName,
+												   queryString);
+	}
+
+	return NIL;
+}
+
+
+/*
+ * ErrorIfUnsupportedAlterTriggerRenameCommand errors out for unsupported
+ * "ALTER TRIGGER oldName ON relationName RENAME TO newName" commands.
+ */
+static void
+ErrorIfUnsupportedAlterTriggerRenameCommand(RenameStmt *renameTriggerStmt)
+{
+	RangeVar *relation = renameTriggerStmt->relation;
+
+	bool missingOk = false;
+	Oid relationId = RangeVarGetRelid(relation, ALTER_TRIGGER_LOCK_MODE, missingOk);
+
+	if (!IsCitusTable(relationId))
+	{
+		return;
+	}
+
+	EnsureCoordinator();
+	ErrorOutForTriggerIfNotCitusLocalTable(relationId);
+}
+
+
+/*
+ * AlterTriggerRenameEventExtendNames extends relation name, old and new trigger
+ * name with shardId, and sets schema name in given RenameStmt.
+ */
+void
+AlterTriggerRenameEventExtendNames(RenameStmt *renameTriggerStmt, char *schemaName,
+								   uint64 shardId)
+{
+	Assert(renameTriggerStmt->renameType == OBJECT_TRIGGER);
+
+	RangeVar *relation = renameTriggerStmt->relation;
+
+	char **relationName = &(relation->relname);
+	AppendShardIdToName(relationName, shardId);
+
+	char **triggerOldName = &(renameTriggerStmt->subname);
+	AppendShardIdToName(triggerOldName, shardId);
+
+	char **triggerNewName = &(renameTriggerStmt->newname);
+	AppendShardIdToName(triggerNewName, shardId);
+
+	char **relationSchemaName = &(relation->schemaname);
+	SetSchemaNameIfNotExist(relationSchemaName, schemaName);
+}
+
+
+/*
+ * PostprocessAlterTriggerDependsStmt is called after a ALTER TRIGGER DEPENDS ON
+ * command has been executed by standard process utility. This function errors out
+ * for unsupported commands or creates ddl job for supported ALTER TRIGGER DEPENDS
+ * ON commands.
+ */
+List *
+PostprocessAlterTriggerDependsStmt(Node *node, const char *queryString)
+{
+	AlterObjectDependsStmt *alterTriggerDependsStmt =
+		castNode(AlterObjectDependsStmt, node);
+	Assert(alterTriggerDependsStmt->objectType == OBJECT_TRIGGER);
+
+	RangeVar *relation = alterTriggerDependsStmt->relation;
+
+	bool missingOk = false;
+	Oid relationId = RangeVarGetRelid(relation, ALTER_TRIGGER_LOCK_MODE, missingOk);
+
+	if (!IsCitusTable(relationId))
+	{
+		return NIL;
+	}
+
+	ErrorIfUnsupportedAlterTriggerDependsCommand(alterTriggerDependsStmt);
```

Comment:
One line before this, we return NIL if the table is not a citus table, but we have the same check in this `ErrorIf...` methods hence they are not covered, I think we can remove one of the same if checks in such methods

Reply:
Actually this part seems to be the same in this method and in the ErrorIf... methods:

```c
	RangeVar *relation = alterTriggerDependsStmt->relation;

	bool missingOk = false;
	Oid relationId = RangeVarGetRelid(relation, ALTER_TRIGGER_LOCK_MODE, missingOk);

	if (!IsCitusTable(relationId))
	{
		return NIL;
	}
```

Reply:
Actually those additional checks in the inner methods (like ErrorIfUnsupportedAlterTriggerDependsCommand) are for the further usage and completeness.
I thought that, anyone that wants to check if an alter trigger command is supported, it should be sufficient to use those functions directly.

Reply:
In that case, why don't we rely on those checks in `ErrorIf`... methods and remove them from callers?

Reply:
```c
	if (!IsCitusTable(relationId))
	{
		return;
	}

	EnsureCoordinator();
	ErrorOutForTriggerIfNotCitusLocalTable(relationId);
```

we do "that check" as we don't want to do `EnsureCoordinator();` for postgres tables.
We also don't want to move `EnsureCoordinator();` after other error checks as @onderkalaci suggested previously.

Reply:
Thinking about this a bit more, I start to think that we can remove this functions.

For example a part from PostprocessCreateTriggerStmt:

```c

	RangeVar *relation = createTriggerStmt->relation;

	bool missingOk = false;
	Oid relationId = RangeVarGetRelid(relation, CREATE_TRIGGER_LOCK_MODE, missingOk);

	if (!IsCitusTable(relationId))
	{
		return NIL;
	}

	ErrorIfUnsupportedCreateTriggerCommand(createTriggerStmt);
```

and `ErrorIfUnsupportedCreateTriggerCommand`:

```c
	RangeVar *relation = createTriggerStmt->relation;

	bool missingOk = false;
	Oid relationId = RangeVarGetRelid(relation, CREATE_TRIGGER_LOCK_MODE, missingOk);

	if (!IsCitusTable(relationId))
	{
		return;
	}

	EnsureCoordinator();
	ErrorOutForTriggerIfNotCitusLocalTable(relationId);
```

So it seems that the `ErrorIfUnsupportedCreateTriggerCommand` is just getting the `relationId` and calling `ErrorOutForTriggerIfNotCitusLocalTable`. We already have the relationId in the caller so we can just do the following in `PostprocessCreateTriggerStmt`:

```c
	RangeVar *relation = createTriggerStmt->relation;

	bool missingOk = false;
	Oid relationId = RangeVarGetRelid(relation, CREATE_TRIGGER_LOCK_MODE, missingOk);

	if (!IsCitusTable(relationId))
	{
		return NIL;
	}

	EnsureCoordinator();
	ErrorOutForTriggerIfNotCitusLocalTable(relationId);
```